### DISCUSSION TITLE
Some node modules are not loaded on production build while using vue-chart-3

### DISCUSSION THREAD

**Discussion by andreas83** [nuxt/nuxt#17935] [Questions]
I use the library vue-chart-3 with nuxt 3 and for some very unknown reason some libs are not loaded in prod build:

Cannot find module .output/server/node_modules/lodash/cloneDeep' imported from .output/server/chunks/app/server.mjs

Not sure if its a vue-chart-3 issue or nuxt3. 

in the mean time i run a patch after every prod build, not sustainable but works for me:

```
@@ -2,11 +2,11 @@
 import dayjs from 'dayjs';
-import debounce from 'lodash/debounce';
+import debounce from 'lodash/debounce.js';
 import * as dateFns from 'date-fns';
 import * as chart from 'chart.js';
-import * as cloneDeep from 'lodash/cloneDeep';
-import * as isEqual from 'lodash/isEqual';
+import * as cloneDeep from 'lodash/cloneDeep.js';
+import * as isEqual from 'lodash/isEqual.js';
 import * as nanoid from 'nanoid';
-import * as camelCase from 'lodash/camelCase';
-import * as startCase from 'lodash/startCase';
+import * as camelCase from 'lodash/camelCase.js';
+import * as startCase from 'lodash/startCase.js';
 import { Html5Qrcode } from 'html5-qrcode';
 import 'unenv/runtime/mock/proxy';
 import 'stream';

```

 ENV
- Operating System: `Linux`
- Node Version:     `v17.1.0`
- Nuxt Version:     `3.0.0-27338323.1e98259`
- Package Manager:  `yarn@1.22.10`
- Bundler:          `Vite`
- User Config:      `buildModules`, `publicRuntimeConfig`
- Runtime Modules:  `-`
- Build Modules:    `nuxt-windicss@2.2.2`

**Comment by naoyoshi** [comment]
In recent days, I'm studying to render Chart.js in Nuxt.js 3.

Firstly, I installed current vue-chart-3 and examined to render at 'npm run dev'.
But it could't work well.
So I changed that I code to render chart directly from Chart.js latest version.
It works well.

A few developers might succeed to render at current vue-chart-3, but I don't know that.
I think that current  Nuxt.js 3 is beta version, so it has many changes till the stable release.
Though vue-chart-3 or vue-chartjs might take much time to respond to Nuxt.js 3,  I wonder that we wait for new version of them.

**Comment by victorgarciaesgi** [comment]
Hi! Maintainer of `vue-chart-3` here. Nuxt 3 being still in beta, it's not really clear what should be updated to work with it yet. I think I need to make an ESM compatible build for Vite/Nuxt 3 but not sure. If the error comes from lodash, maybe I need to add lodash-es but it could break compatibility with old versions of Vue or for non-Vite projects

**Reply by toniengelhardt** [reply]
Is there maybe a way to manually add lodash-es when using it within Nuxt 3 to temporarily fix this  issue?

**Reply by toniengelhardt** [reply]
`vue-chart-3` works fine with Nuxt 3 with the dev server (except reactivity in my case), only the build creates problems.

**Comment by toniengelhardt** [comment]
@andreas83 may I ask how you patch this bug?

**Reply by andreas83** [reply]
I modified the generated source `.output/server/chunks/app/server.mjs`

just add .js to the imports of lodash i.e. 

```
import * as cloneDeep from 'lodash/cloneDeep';
```

to 
```
import * as cloneDeep from 'lodash/cloneDeep.js';
```

hope this helps.

**Reply by andreas83** [reply]
@victorgarciaesgi was so kind and updated his lib, so it should work now with vue-chart 3.1.0, can you confirm?

**Reply by toniengelhardt** [reply]
Amazing, thanks!
